crates/telio-nurse/src/qos.rs

```diff
@@ -181,9 +199,20 @@ impl Runtime for Analytics {
                 })
             },
 
-            _ = self.rtt_interval.tick() => {
+            _ = self.rtt_interval.tick(), if self.ping_channel_tx.upgrade().is_none() => {
+                Self::guard(async move {
+                    self.perform_ping();
+                    Ok(())
+                })
+            },
+
+            Some(node_ping_res) = self.ping_channel_rx.recv() => {
                 Self::guard(async move {
-                    self.perform_ping().await;
+                    self.nodes
+                        .entry(node_ping_res.public_key)
+                        .and_modify(|n| {
+                            *n = node_ping_res;
```

Comment:
I think making the pinger send only the ping results is safer.
This thing can destroy good data if something bad happens in the pinger.

Reply:
Agreed, will change